grovedb/src/lib.rs

```diff
@@ -274,6 +283,49 @@ impl GroveDb {
         Ok(proof_result)
     }
 
+    pub fn verify_proof(
+        path: &[&[u8]],
+        proofs: Vec<Vec<u8>>,
+        expected_root_hash: [u8; 32],
+    ) -> Result<Map, Error> {
+        // Should it really be 2 or more??
+        if proofs.len() < 2 {
+            return Err(Error::InvalidProof("Proof length should be 2 or more"));
+        }
+
+        if proofs.len() - 1 != path.len() {
+            return Err(Error::InvalidProof("Proof length should be one greater than path"));
+        }
+
+        let (mut last_root_hash, mut last_result_map) = execute_proof(&proofs[0][..]).unwrap();
+
+        for i in 1..proofs.len() {
+            if i == proofs.len() - 1 {
+                // Last proof (root proof)
+                let root_proof = MerkleProof::<Sha256>::try_from(&proofs[i][..]).unwrap();
+                let a: [u8; 32] = last_root_hash;
+                if root_proof.verify(expected_root_hash, &vec![0], &[a], 2) {
+                    break;
+                } else {
+                    return Err(Error::InvalidProof("Root hashes didn't match"));
+                }
+            } else {
+                // Merk proof, validate that the proof is valid and
+                // the result map contains the last hash i.e the previous
+                // merk was a child of this merk
+                let proof_result = execute_proof(&proofs[i][..]).unwrap();
```

Comment:
If these `unwrap`s cannot panic please use `expect(..)` with a description why the won't do so, but I suppose these may panic